otter/test/test_metrics.py

```diff
@@ -447,6 +422,127 @@ def test_diverged(self):
             [(metrics[1], 3603), (metrics[2], 7203)])
 
 
+class GetTodaysTenants(SynchronousTestCase):
+    """
+    Tests for :func:`get_todays_tenants`
+    """
+
+    def setUp(self):
+        self.tenants = range(10)
+        self.today = datetime(1970, 1, 2)
+
+    def test_last_none(self):
+        """
+        returns first 5 sorted tenants with length 5 and todays date
+        """
+        self.assertEqual(
+            get_todays_tenants(self.tenants, self.today, None, None),
+            (self.tenants[:5], 5, self.today))
+
+    def test_same_day(self):
+        """
+        returns same tenants as last time if asked within same day
+        """
+        last_date = self.today.replace(hour=13, minute=20)
+        self.assertEqual(
+            get_todays_tenants(self.tenants, self.today, 3, last_date),
+            (self.tenants[:3], 3, self.today))
+
+    def test_next_day(self):
+        """
+        returns tenants with 5 more for next day
+        """
+        prev_day = datetime(1970, 1, 1)
+        self.assertEqual(
+            get_todays_tenants(self.tenants, self.today, 3, prev_day),
+            (self.tenants[:8], 8, self.today))
+
+    def test_all(self):
+        """
+        returns all tenants for new day if < 5 tenants are remaining since
+        last time
+        """
+        prev_day = datetime(1970, 1, 1)
+        self.assertEqual(
+            get_todays_tenants(self.tenants, self.today, 7, prev_day),
+            (self.tenants, 10, self.today))
+
+    def test_previous_day(self):
+        """
+        Same as `test_same_day`
+        """
+        next_day = datetime(1970, 1, 3)
+        self.assertEqual(
+            get_todays_tenants(self.tenants, self.today, 3, next_day),
+            (self.tenants[:3], 3, self.today))
+
+
+class GetTodaysScalingGroupsTests(SynchronousTestCase):
+    """
+    Tests for :func:`get_todays_scaling_groups`
+    """
+
+    def setUp(self):
+        self.groups = groupby(
+            lambda g: g["tenantId"],
+            ([{"tenantId": "t1", "a": "1"}, {"tenantId": "t1", "a": "2"}] +
+             [{"tenantId": "t{}".format(i), "b": str(i)}
+              for i in range(2, 10)]))
+
+    def test_success(self):
+        """
+        Returns todays scaling groups based on number of tenants fetched
+        since last time. Updates the current fetch in file
+        """
+        seq = [
```

Comment:
So neat :D  I love that it tests both the convergence IDs + the additional tenants in each test.